cpp/include/cudf/strings/strings_column_factories.hpp

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cudf/null_mask.hpp>
+#include <cudf/types.hpp>
+
+#include <rmm/thrust_rmm_allocator.h>
+#include <cudf/column/column.hpp>
+
+namespace cudf {
+/**---------------------------------------------------------------------------*
+ * @brief Construct strings column given an array of pointer/size pairs.
+ * Use the strings_column_handler class to perform strings operations on
+ * this type of column.
+ * 
+ * @note `null_count()` and `null_bitmask` are determined if a pair contains
+ * a null pointer. Otherwise, it is considered an empty string and not null.
```

Comment:
Make a little more explicit:
- `char* == nullptr` -> NULL string for all values of `size`
- `size==0 and char* != nullptr` -> non-null, empty string